File: frontend/src/services/auth.service.js

// d:\sunnyshop\frontend\src\services\auth.service.js
import api from '@/config/api';

export const authService = {
  login: async (username, password) => {
    const response = await api.post('/auth/login', { username, password });
    return response.data;
  },

  register: async (data) => {
    const response = await api.post('/auth/register', data);
    return response.data;
  },

  logout: async () => {
    const response = await api.post('/auth/logout');
    return response.data;
  },
  
  forgotPassword: async (email) => {
     const response = await api.post('/auth/forgot-password', { email });
     return response.data;
  },

  verifyOtp: async (otp, username) => {
     const response = await api.post('/auth/verify-otp', { otp, username });
     return response.data;
  },
  
  resetPassword: async (token, newPassword) => {
     const response = await api.post('/auth/reset-password', { token, newPassword });
     return response.data;
  }
};

File: api-server/src/main/java/com/sonnguyen/laptopshop/service/AuthService.java
package com.sonnguyen.laptopshop.service;

import com.sonnguyen.laptopshop.model.User;
import com.sonnguyen.laptopshop.payload.request.RegisterRequest;
import com.sonnguyen.laptopshop.payload.response.AuthResponse;

public interface AuthService {
    AuthResponse login(String username, String password);
    void register(RegisterRequest registerRequest);
    User getCurrentUser(String username);
    User updateProfile(String username, RegisterRequest updateRequest);
    void logout(String username);
    AuthResponse refreshToken(String refreshToken);
    void changePassword(String username, String oldPassword, String newPassword);
    void forgotPassword(String email);
    void verifyOtp(String otp, String username);
    void resetPassword(String token, String newPassword);
}
